src/applications/virtual-agent/hooks/useChatbotToken.js

```diff
@@ -29,17 +39,38 @@ async function getToken(setToken, setCode, setLoadingStatus) {
   }
 }
 
-export default function useChatbotToken(props) {
+export default function useChatbotToken() {
   const [token, setToken] = useState('');
   const [code, setCode] = useState('');
   const [loadingStatus, setLoadingStatus] = useState(LOADING);
+  const { useToggleValue, TOGGLE_NAMES } = useFeatureToggle();
+  const isSessionPersistenceEnabled = useToggleValue(
+    TOGGLE_NAMES.virtualAgentChatbotSessionPersistenceEnabled,
+  );
 
   useEffect(
     () => {
+      // Runs on mount and whenever isSessionPersistenceEnabled changes: prevents duplicate fetches on re-renders
       clearBotSessionStorage();
+
+      // When persistence is OFF, force fresh token/conversationId.
+      if (!isSessionPersistenceEnabled) {
+        getToken(setToken, setCode, setLoadingStatus, true);
+        return;
+      }
```

Comment:
Do you think it would be worth checking for the toggle loading state here as well? I was thinking that if the toggles are slow to load there may be a false positive here that would result in the getToken getting called.